test/distributed/pipelining/test_stage.py

```diff
@@ -257,6 +257,107 @@ def _run_step(x):
             with self.assertRaisesRegex(PipeliningShapeError, "dtype mismatch"):
                 _run_step(x)
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_custom_dw_with_fb_schedule(self):
+        """Tests that separate weight grad function 'dw_runner' gets run under a schedule that's only aware of F/B."""
+        full_mod = MultiMLP(d_hid, n_layers=self.world_size)
+        full_mod.to(self.device)
+        stage_mod = full_mod.get_submodule(f"layers.{self.rank}")
+
+        x = torch.randn(batch_size, d_hid, device=self.device)
+        target = torch.randn(batch_size, d_hid, device=self.device)
+
+        class CustomState:
+            def __init__(self):
+                self.i = 0
+
+            def dw_builder(self):
```

Comment:
Do you have a reference/code pointers to how the custom backward is currently implemented in xlformers? How would they use this API on their end